release-common-lib/src/main/java/org/reactome/util/compare/DBObjectComparer.java

```diff
@@ -22,27 +26,34 @@
  */
 public class DBObjectComparer
 {
+	private static Map<GKInstance, Map<SchemaAttribute, List<Object>>> instanceAttributeToValuesMap = new HashMap<>();
+	private static Map<GKInstance, List<SchemaAttribute>> instanceToRegularAttributesMap = new HashMap<>();
+	private static Map<GKInstance, List<SchemaAttribute>> instanceToReferrerAttributesMap = new HashMap<>();
+
 	/**
 	 * Compares two GKInstances.
-	 * @param inst1 the first instance.
-	 * @param inst2 the second instance.
-	 * @param sb a StringBuilder that will contain a detailed report of differences.
+	 * @param instance1 The first instance.
+	 * @param instance2 The second instance.
+	 * @param stringBuilder A StringBuilder that will contain a detailed report of differences.
 	 * @return The number of differences between the two instances.
 	 * A single-valued attribute that differs will count as 1 diff.
 	 * If the instances have different schema classes, that will count as 1 diff.
 	 * If a multi-valued attribute has a different number of elements between the two instances,
 	 * that will count as 1 diff and the elements will NOT be compared.
 	 */
-	public static int compareInstances(GKInstance inst1, GKInstance inst2, StringBuilder sb)
+	public static int compareInstances(GKInstance instance1, GKInstance instance2, StringBuilder stringBuilder)
 	{
-		return compareInstances(inst1, inst2, sb, 5, false);
+		final int maxRecursionDepth = 5;
+		final boolean checkReferrers = false;
```

Comment:
Should this be a global variable and capitalized? I don't personally have an issue with this way of doing it, but should it be that way for consistency across the project?

Reply:
Usually, it is only moved to class-level scope and CAPITALIZED if it's going to be accessed in several different places.

Reply:
I think for this particular case there was an additional  time where it was accessed, in the function immediately following it

Reply:
Yup, you're right.  I will move these to be class-scoped since they are re-used.